ext/travisci/prep-os-essentials-for

```diff
@@ -11,6 +11,24 @@ misuse() {
     exit 2
 }
 
+# this handles the issue where apt-get update fails but has an exit code of zero
+# because there's a mirror sync going on. This failure is retryable, but you have
+# to look for it in the output from the command. This is a common failure on AWS
+# and GCP because they inject their own mirrors into /etc/apt/sources.list at VM
+# start time, and there's a delay in getting all their mirrors in sync.
+apt_get_update() {
+    local tmpdir
+    tmpdir=$(mktemp)
+    set +e
+    for i in $(seq 1 20); do
+        sudo apt-get update 2>&1 | tee $tmpdir
+        grep -x "[WE]:.*" $tmpdir || break
+        sleep 1
+    done;
+    set -e
+    echo Executed apt-get update $i times;
+}
```

Comment:
Noticed a stray semicolon after the done, suspect `tmpdir` should perhaps be `tmpfile` and I'd prefer to narrow the `set +e` use if we can, since for example, this won't notice a real apt-get, sleep, or tee failure (though maybe we don't care about apt-get failures here?).

In any case, with the current structure, I'm not sure we need the `set +e/-e` anyway, unless we want to keep going after real apt-get failures.  (If so, perhaps we could just do something like this: `(sudo apt-get update 2>&1 || true) | tee "$tmpdir"`.  (...I'd quote the tmpdir expansions so we don't have to care what mktemp does.)